mne/encoding/tests/test_encoding.py

```diff
@@ -0,0 +1,181 @@
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#
+# License: BSD (3-clause)
+import warnings
+import os.path as op
+
+from nose.tools import assert_raises, assert_true, assert_equal
+import numpy as np
+from numpy.testing import assert_array_equal
+
+from mne import io, read_events, pick_types
+from mne.utils import (requires_sklearn, run_tests_if_main)
+
+
+data_dir = op.join(op.dirname(__file__), '..', '..', 'io', 'tests', 'data')
+raw_fname = op.join(data_dir, 'test_raw.fif')
+event_name = op.join(data_dir, 'test-eve.fif')
+
+np.random.seed(1337)
+
+tmin, tmax = -0.1, 0.5
+event_id = dict(aud_l=1, vis_l=3)
+
+warnings.simplefilter('always')
+
+# Loading raw data
+raw = io.read_raw_fif(raw_fname, preload=True)
+picks = pick_types(raw.info, meg=True, stim=False, ecg=False,
+                   eog=False, exclude='bads')
+picks = picks[0:2]
+
+
+@requires_sklearn
+def test_feature():
+    from mne.encoding import (SampleMasker, EventsBinarizer,
+                              FeatureDelayer)
+    from sklearn.linear_model import Ridge
+    from scipy.sparse import csr_matrix
+    events = read_events(event_name)
+    events[:, 0] -= raw.first_samp
+
+    # --- Feature Delayer ---
+    # Explicit delays + sfreq
+    X = np.random.randn(1000, 2)
+    for idel, isfreq in [[(0, 1, 2), 1], [(0, .1, .2), 10]]:
+        delayer = FeatureDelayer(delays=idel, sfreq=isfreq)
+        Xdel = delayer.transform(X)
+        assert_array_equal(Xdel[:, 0], X[:, 0])
+        assert_array_equal(Xdel[:-1, 1], X[1:, 0])
```

Comment:
it just makes sure that the 1st column of Xdel is in fact the correctly shifted version of the first column of X